### PR TITLE
MQE-2298: Move and update test BIC document to DevDocs

### PR DESCRIPTION
**Pull request by jilu1**
## Purpose of this pull request

This pull request (PR) moves MFTF versioning topic to extension dev guide.

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.4/extension-dev-guide/versioning/mftf-tests-codebase-changes.html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Please add an appropriate description of the PR as per DevDocs guidelines
Correct linting errors.

Also this is a new page but there is no menu point to get to this page - this should be created.

Reviewer assumes that new technical content is correct.

File: src/guides/v2.4/extension-dev-guide/versioning/mftf-tests-codebase-changes.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    X.Y.Z
+    | | |
+    | | +-- Backward Compatible changes (bug fixes)
+    | +---- Backward Compatible changes (new features)
+    +------ Backward Incompatible changes
+
+### Z release
+
+Patch version **Z** MUST be incremented if only backward compatible changes to tests are introduced.
+For instance: a fix which aims to resolve test flakiness. This can be done by updating an unreliable selector, adding a `wait` to an element, or updating a data entity value.
```
[the remark this answers is not in the document]
Comment (dobooth) [reply src/guides/v2.4/extension-dev-guide/versioning/mftf-tests-codebase-changes.md:42]: I think "flakiness" is almost a  term of art in testing.

**Comment by dobooth** [comment]
running tests

**Review by dobooth — APPROVED** [review]

**Pull request merged**